src/evaluate/hub.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict
+
+import requests
+from datasets.utils.metadata import known_task_ids
+from huggingface_hub import dataset_info, model_info
+from huggingface_hub.repocard import metadata_update
+
+from .utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+def get_allowed_tasks(tasks_dict):
+    return list(tasks_dict.keys())
+
+
+def push_to_hub(
+    repo_id: str,
+    task_type: str,
+    dataset_type: str,
+    dataset_name: str,
+    metric_type: str,
+    metric_name: str,
+    metric_value: float,
+    task_name: str = None,
+    dataset_config: str = None,
+    dataset_split: str = None,
+    dataset_revision: str = None,
+    dataset_args: Dict[str, int] = None,
+    metric_config: str = None,
+    metric_args: Dict[str, int] = None,
+    overwrite: bool = False,
+):
+    r"""
+    Pushes the result of a metric to the Hub, adding it to the specified model's card.
+
+    Args:
+        repo_id (``str``): Model id from https://hf.co/datasets.
```

Comment:
Here the users really work with 3 repos: a dataset repo, a model repo and a metrics repo, so I think this could lead to confusion. Would it make more sense to have model_id, dataset_id, and metric_id?

Reply:
Sure, I can totally rename `repo_id` to `model_id`! For the dataset and metric though, I'd called them `*_type` because that's what they're called in the [model card spec](https://raw.githubusercontent.com/huggingface/hub-docs/main/modelcard.md). Wouldn't it be confusing for the params to mismatch?

Reply:
You're right, consistency is probably better!